### ISSUE TITLE
Rot3 has a wired twofold implementation

### ISSUE THREAD

**Issue by relleums** [cherenkov-plenoscope/merlict_development_kit#7]
The Rot3 class represents Tait-Bryan angles in a 3D space. Together with Vec3 it forms the homogeneous transformation HomTra3. However, it has a wired twofold behaviour. It can store the Tait-Bryan angles OR a rotation axis and its associated rotation angle. Rot3 further has a flag which indicates which of the two representation it stores. Based on this flag, one has to call different members of Rot3 to work with it. This stinks, and must be re factored.
We could:
* create a Tait-Bryan and a separate rot-axis class
* Drop both concepts (Tait-Bryan and rot axis) and only have a rotation class which stores a orthogonal rotation matrix. The HomTra3D creates this rotation matrix anyhow from both the two concepts when given a Rot3.

**Comment by dneise** [comment]
Is this ultimately a question of user-facing interface? So let's say you decide for point 2 (Drop both and just store orth. rot. matrix), then I assume it would still be nice to have something like:

```python
my_rot = Rot3.from_tait_bryan(yaw, pitch, roll)
my_other_rot = Rot3.from_euler(a,b,c)
my_best_rot = Rot3.from_dir_n_angle(direction_vector, angle)
```
Right?

**Comment by dneise** [comment]
Btw. I totally agree, having this part:

> It can store the Tait-Bryan angles OR a rotation axis and its associated rotation angle. Rot3 further has a flag which indicates which of the two representation it stores.

smells ... It looks somehow, like a lazy evaluation, right? Instead of immediately storing an orth. rot. matrix, it stores exactly what it got from the user and remembers how it was made ... and only later is converted into a rotational matrix...

**Comment by relleums** [comment]
It is a question of the user-interface, and a little bit about the internal usage.

Users shall be able to define rotations in different ways:
- tait-biyan-angles
- rot-axis, and rot-angle
- maybe even euler-angles

On the inner side is the 6-degrees-of-freedom Transformation between each two frames.
In merlicht this is implemented using the homogeneous-transformation-matrix  ```HomTra3```, but could also be implemented using e.g. Quarternions like in FACT-MARS.

It is a little bit the question when the user-defined rotations shall be evaluated. The ray-tracer only needs the resulting 6-degrees-of-freedom Transformation. This is somewhat coupled to issue #10 where the Frame has too much responsibility.

I somehow think, there shall be structs for representing the scenery as the user wants it, and a second set of structs which represent the scenery as the ray-tracer needs it to be efficient. E.g. [automatic creation of non user-defined volume-bound-hierarchies](https://github.com/cherenkov-plenoscope/merlict_development_kit/blob/9f2965efb66830097e5a761f9a7598d205ba85e1/merlict/Frame.cpp#L190).
The first set of structs can have either tait-biyan-angles, or rot-axis, and rot-angle, where the second set of structs only stores the 6-degrees-of-freedom Transformation.

**Comment by dneise** [comment]
I totally agree. Even more, I think this can solve a little issue, a typical propagation program currently has. I mean, I speak of "typical" but actually my overview is rather limited.... anyway.

Currently I see this as a [typical propagation routine](https://github.com/cherenkov-plenoscope/merlict_development_kit/blob/770cd2769c5a36585a498ec2ee1ae5eb2dd17cb9/merlict/tests/PhotonTest.cpp#L84)

1. setup `scenery`, i.e. colors, functions, and the scenery itself.
2. call  `scenery.root.init_tree_based_on_mother_child_relations()`;
3. create `photons`
4. call `propagate_photons_in_frame(photons, scenery)`;
5. evaluate the results by looking at photons, i.e. most ofthen their `final_intersection`.

I think here step 2 can easily be forgotten, since it modifies the scenery in place.

---

Now if we think of your proposal here, where one has one scenery as the user wants it and one scenery as the ray tracer needs it, then a typical procedure might be:


1. setup `scenery`, i.e. colors, functions, and the scenery itself.
2. create `photons`
3. call `propagate_photons_in_frame(photons, scenery)`;
4. evaluate the results by looking at photons, i.e. most ofthen their `final_intersection`.

and the conversion of sceneries is done within the `propagate` call...
